### PR TITLE
(PDB-5613) Remove PSON

### PR DESCRIPTION
**Pull request by austb**

### CODE REVIEW

File: puppet/lib/puppet/util/puppetdb/command.rb [hunk 1]
```diff
@@ -41,14 +63,14 @@ def initialize(command, version, certname, producer_timestamp_utc, payload)
       #
       # This is roughly inline with how Puppet serializes for catalogs as of
       # Puppet 4.1.0. We need a better answer to non-utf8 data end-to-end.
```
Comment (justinstoller) [inline puppet/lib/puppet/util/puppetdb/command.rb:65]: The future is here, this comment is out of date!

File: puppet/lib/puppet/util/puppetdb/command.rb [hunk 2]
```diff
@@ -41,14 +63,14 @@ def initialize(command, version, certname, producer_timestamp_utc, payload)
...
-      }.to_pson, "Error encoding a '#{command}' command for host '#{certname}'")
+      }.to_json, "Error encoding a '#{command}' command for host '#{certname}'")
     end
     @checksum = Digest::SHA1.hexdigest(checksum_payload)
     @command = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(command).gsub(" ", "_")
     @version = version
     @certname = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(certname)
     @producer_timestamp_utc = producer_timestamp_utc
-    @payload = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(payload.to_pson)
+    @payload = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(coerced_payload.to_json)
```
Comment (justinstoller) [inline puppet/lib/puppet/util/puppetdb/command.rb:73]: effectively calling `to_json` on the payload twice seems wasteful.

**Comment by jpartlow** [comment]
Looks like it wants the puppet-pson gem for the puppetlabs.puppetdb.integration.file-with-binary-template/file-with-binary-template test.

File: puppet/lib/puppet/util/puppetdb/command.rb [hunk 3]
```diff
@@ -11,6 +11,53 @@ class Puppet::Util::Puppetdb::Command
...
+  # @param payload Initially the whole Puppet object converted
+  #   to_data_hash. The called recusively if the command has neseted
+  #   structures.
+  # @param error_context_str a string prefix for log messages
+  def coerce_payload(payload, error_context_str)
+    c = ""
+    had_lossy_string = false
+    case payload
+    when Hash
+      c = payload.map do |k, v|
```
Comment (jpartlow) [inline puppet/lib/puppet/util/puppetdb/command.rb:25]: Could also use each_with_object({}) here.

File: puppet/lib/puppet/util/puppetdb/char_encoding.rb [hunk 4]
```diff
@@ -64,7 +43,9 @@ def self.first_invalid_char_range(str)
   # @param str string coming from to_pson, likely a command to be submitted to PDB
   # @param bad_char_range a range indicating a block of invalid characters
```
Comment (jpartlow) [inline puppet/lib/puppet/util/puppetdb/char_encoding.rb:44]: This parameter doesn't exist anymore.

File: puppet/lib/puppet/util/puppetdb/char_encoding.rb [hunk 5]
```diff
@@ -109,7 +66,7 @@ def self.warn_if_invalid_chars(str, error_context_str)
   # @param error_context_str information about where this string came from for
```
Comment (jpartlow) [inline puppet/lib/puppet/util/puppetdb/char_encoding.rb:66]: This parameter is gone.

File: puppet/lib/puppet/util/puppetdb/char_encoding.rb [hunk 6]
```diff
@@ -76,30 +57,6 @@ def self.error_char_context(str, bad_char_range)
...
-        Puppet.debug error_context_str + "\n" + error_char_context(str, first_invalid_char_range(str))
-      end
-
-      str
-    end
-  end
-
   # @api private
   #
   # Attempts to coerce str to UTF-8, if that fails will output context
```
Comment (jpartlow) [inline puppet/lib/puppet/util/puppetdb/char_encoding.rb:62]: And the description is referencing error_context_str as well.

File: puppet/lib/puppet/util/puppetdb/command.rb [hunk 7]
```diff
@@ -11,6 +11,53 @@ class Puppet::Util::Puppetdb::Command
 
   CommandsUrl = "/pdb/cmd/v1"
 
+  # Recursively coerce all the strings in a payload to UTF-8
+  #
+  # @param payload Initially the whole Puppet object converted
+  #   to_data_hash. The called recusively if the command has neseted
```
Comment (jpartlow) [inline puppet/lib/puppet/util/puppetdb/command.rb:17]: Then called? ... nested

File: puppet/lib/puppet/util/puppetdb/command.rb (hunk 3 above)
Comment (donoghuc) [reply puppet/lib/puppet/util/puppetdb/command.rb]: updated

**Comment by donoghuc** [comment]
https://tickets.puppetlabs.com/browse/PDB-5622 for the failing aceptance test. It has been red for a long time. I think we should get this merged and address the acceptance test issue separately.

**Pull request merged**